eclair-core/src/main/scala/fr/acinq/eclair/router/Announcements.scala

```diff
@@ -76,15 +76,16 @@ object Announcements {
       case address@(_: Tor2) => (3, address)
       case address@(_: Tor3) => (4, address)
     }.sortBy(_._1).map(_._2)
-    val witness = nodeAnnouncementWitnessEncode(timestamp, nodeSecret.publicKey, color, alias, features, sortedAddresses, TlvStream.empty)
+    val nodeAnnouncementFeatures = features.nodeAnnouncementFeatures()
+    val witness = nodeAnnouncementWitnessEncode(timestamp, nodeSecret.publicKey, color, alias, nodeAnnouncementFeatures, sortedAddresses, TlvStream.empty)
     val sig = Crypto.sign(witness, nodeSecret)
     NodeAnnouncement(
       signature = sig,
       timestamp = timestamp,
       nodeId = nodeSecret.publicKey,
       rgbColor = color,
       alias = alias,
-      features = features,
+      features = nodeAnnouncementFeatures,
```

Comment:
Agreed, typing those becomes a bit messy because we can upcast `Map[Feature with NodeFeature, FeatureSupport]` to `Map[Feature, FeatureSupport]` so we'd need conversions...